### YamlMime:ManagedReference
items:
  - uid: com.microsoft.azure.eventhubs._event_hub_exception
    id: _event_hub_exception
    parent: com.microsoft.azure.eventhubs
    children:
      - 'com.microsoft.azure.eventhubs._event_hub_exception.EventHubException(final boolean,final String)'
      - 'com.microsoft.azure.eventhubs._event_hub_exception.EventHubException(final boolean,final String,final Throwable)'
      - 'com.microsoft.azure.eventhubs._event_hub_exception.EventHubException(final boolean,final Throwable)'
      - com.microsoft.azure.eventhubs._event_hub_exception.getContext()
      - com.microsoft.azure.eventhubs._event_hub_exception.getIsTransient()
      - com.microsoft.azure.eventhubs._event_hub_exception.getMessage()
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: EventHubException
    nameWithType: EventHubException
    fullName: com.microsoft.azure.eventhubs.EventHubException
    type: Class
    source:
      remote: &ref_0
        path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
        branch: master
        repo: 'https://github.com/Azure/azure-event-hubs-java'
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 11
    package: com.microsoft.azure.eventhubs
    summary: '<p><span data-ttu-id="760e8-101">Dies ist die Basis-Ausnahme, die für alle Fehlerfälle Servicebus Sequenzelement erzeugt.</span><span class="sxs-lookup"><span data-stu-id="760e8-101">This is the base exception that service bus will produce for all error cases.</span></span> </p>'
    syntax:
      content: public class EventHubException
    inheritance:
      - java.lang.Object
      - Exception
  - uid: 'com.microsoft.azure.eventhubs._event_hub_exception.EventHubException(final boolean,final String)'
    id: 'EventHubException(final boolean,final String)'
    parent: com.microsoft.azure.eventhubs._event_hub_exception
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: 'EventHubException(final boolean isTransient, final String message)'
    nameWithType: 'EventHubException.EventHubException(final boolean isTransient, final String message)'
    fullName: 'com.microsoft.azure.eventhubs.EventHubException.EventHubException(final boolean isTransient, final String message)'
    overload: com.microsoft.azure.eventhubs._event_hub_exception.EventHubException*
    type: Constructor
    source:
      remote: *ref_0
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 24
    package: com.microsoft.azure.eventhubs
    syntax:
      content: 'public EventHubException(final boolean isTransient, final String message)'
      parameters:
        - id: isTransient
          type: 4dda0764
        - id: message
          type: 3d93591e
  - uid: 'com.microsoft.azure.eventhubs._event_hub_exception.EventHubException(final boolean,final String,final Throwable)'
    id: 'EventHubException(final boolean,final String,final Throwable)'
    parent: com.microsoft.azure.eventhubs._event_hub_exception
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: 'EventHubException(final boolean isTransient, final String message, final Throwable cause)'
    nameWithType: 'EventHubException.EventHubException(final boolean isTransient, final String message, final Throwable cause)'
    fullName: 'com.microsoft.azure.eventhubs.EventHubException.EventHubException(final boolean isTransient, final String message, final Throwable cause)'
    overload: com.microsoft.azure.eventhubs._event_hub_exception.EventHubException*
    type: Constructor
    source:
      remote: *ref_0
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 36
    package: com.microsoft.azure.eventhubs
    syntax:
      content: 'public EventHubException(final boolean isTransient, final String message, final Throwable cause)'
      parameters:
        - id: isTransient
          type: 4dda0764
        - id: message
          type: 3d93591e
        - id: cause
          type: fa00b19f
  - uid: 'com.microsoft.azure.eventhubs._event_hub_exception.EventHubException(final boolean,final Throwable)'
    id: 'EventHubException(final boolean,final Throwable)'
    parent: com.microsoft.azure.eventhubs._event_hub_exception
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: 'EventHubException(final boolean isTransient, final Throwable cause)'
    nameWithType: 'EventHubException.EventHubException(final boolean isTransient, final Throwable cause)'
    fullName: 'com.microsoft.azure.eventhubs.EventHubException.EventHubException(final boolean isTransient, final Throwable cause)'
    overload: com.microsoft.azure.eventhubs._event_hub_exception.EventHubException*
    type: Constructor
    source:
      remote: *ref_0
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 30
    package: com.microsoft.azure.eventhubs
    syntax:
      content: 'public EventHubException(final boolean isTransient, final Throwable cause)'
      parameters:
        - id: isTransient
          type: 4dda0764
        - id: cause
          type: fa00b19f
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.getContext()
    id: getContext()
    parent: com.microsoft.azure.eventhubs._event_hub_exception
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: getContext()
    nameWithType: EventHubException.getContext()
    fullName: ErrorContext com.microsoft.azure.eventhubs.EventHubException.getContext()
    overload: com.microsoft.azure.eventhubs._event_hub_exception.getContext*
    type: Method
    source:
      remote: *ref_0
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 62
    package: com.microsoft.azure.eventhubs
    syntax:
      content: public ErrorContext getContext()
      return:
        type: com.microsoft.azure.eventhubs._error_context
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.getIsTransient()
    id: getIsTransient()
    parent: com.microsoft.azure.eventhubs._event_hub_exception
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: getIsTransient()
    nameWithType: EventHubException.getIsTransient()
    fullName: boolean com.microsoft.azure.eventhubs.EventHubException.getIsTransient()
    overload: com.microsoft.azure.eventhubs._event_hub_exception.getIsTransient*
    type: Method
    source:
      remote: *ref_0
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 57
    package: com.microsoft.azure.eventhubs
    summary: '<p><span data-ttu-id="760e8-102">Ein boolescher Wert, der angibt, ob die Ausnahme ein vorübergehender Fehler ist.</span><span class="sxs-lookup"><span data-stu-id="760e8-102">A boolean indicating if the exception is a transient error or not.</span></span> </p>'
    syntax:
      content: public boolean getIsTransient()
      return:
        type: 4fc6e284
        description: '<p><span data-ttu-id="760e8-103">Gibt "true" zurück, wenn der Benutzer den Vorgang wiederholen kann, der die Ausnahme ohne weiteren Eingriff generiert hat.</span><span class="sxs-lookup"><span data-stu-id="760e8-103">returns true when user can retry the operation that generated the exception without additional intervention.</span></span> </p>'
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.getMessage()
    id: getMessage()
    parent: com.microsoft.azure.eventhubs._event_hub_exception
    href: com.microsoft.azure.eventhubs._event_hub_exception.yml
    langs:
      - java
    name: getMessage()
    nameWithType: EventHubException.getMessage()
    fullName: String com.microsoft.azure.eventhubs.EventHubException.getMessage()
    overload: com.microsoft.azure.eventhubs._event_hub_exception.getMessage*
    type: Method
    source:
      remote: *ref_0
      path: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubException.java
      startLine: 43
    package: com.microsoft.azure.eventhubs
    syntax:
      content: public String getMessage()
      return:
        type: '26831127'
references:
  - uid: 4dda0764
    spec.java:
      - name: final boolean
        fullName: final boolean
  - uid: 3d93591e
    spec.java:
      - name: final String
        fullName: final String
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.EventHubException*
    name: EventHubException
    nameWithType: EventHubException.EventHubException
    fullName: com.microsoft.azure.eventhubs.EventHubException.EventHubException
    package: com.microsoft.azure.eventhubs
  - uid: fa00b19f
    spec.java:
      - name: final Throwable
        fullName: final Throwable
  - uid: '26831127'
    spec.java:
      - name: String
        fullName: String
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.getMessage*
    name: getMessage
    nameWithType: EventHubException.getMessage
    fullName: String com.microsoft.azure.eventhubs.EventHubException.getMessage
    package: com.microsoft.azure.eventhubs
  - uid: 4fc6e284
    spec.java:
      - name: boolean
        fullName: boolean
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.getIsTransient*
    name: getIsTransient
    nameWithType: EventHubException.getIsTransient
    fullName: boolean com.microsoft.azure.eventhubs.EventHubException.getIsTransient
    package: com.microsoft.azure.eventhubs
  - uid: com.microsoft.azure.eventhubs._event_hub_exception.getContext*
    name: getContext
    nameWithType: EventHubException.getContext
    fullName: ErrorContext com.microsoft.azure.eventhubs.EventHubException.getContext
    package: com.microsoft.azure.eventhubs
  - uid: com.microsoft.azure.eventhubs._authorization_failed_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._authorization_failed_exception.yml
    name: AuthorizationFailedException
    nameWithType: AuthorizationFailedException
    fullName: com.microsoft.azure.eventhubs.AuthorizationFailedException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-104">Autorisierung Ausnahmefehler ausgelöst wird, wenn Fehler aufgetreten ist, während die Benutzerberechtigungen zum Ausführen der gewünschten Vorgänge autorisieren.</span><span class="sxs-lookup"><span data-stu-id="760e8-104">Authorization failed exception is thrown when error is encountered during authorizing user's permission to run the intended operations.</span></span> <span data-ttu-id="760e8-105">Wenn diese Ausnahme festgestellt Benutzer sollte überprüfen Sie, ob die Token-Schlüssel in der Verbindungszeichenfolge angegeben (z. B. mit dem übergebenen <xref uid="com.microsoft.azure.eventhubs._event_hub_client.createFromConnectionString(final String)" data-throw-if-not-resolved="false">com.microsoft.azure.eventhubs.EventHubClient#createFromConnectionString(String)</xref>) gültig ist, und verfügt über die korrekte Ausführung Recht für die vorgesehenen Vorgänge (z. B. empfangsaufruf Anspruch zugeordnete die Schlüssel bzw. das Token Lauschen müssen).</span><span class="sxs-lookup"><span data-stu-id="760e8-105">When encountered this exception user should check whether the token/key provided in the connection string (e.g. one passed to <xref uid="com.microsoft.azure.eventhubs._event_hub_client.createFromConnectionString(final String)" data-throw-if-not-resolved="false">com.microsoft.azure.eventhubs.EventHubClient#createFromConnectionString(String)</xref>) is valid, and has correct execution right for the intended operations (e.g. Receive call will need Listen claim associated with the key/token).</span></span></p>
      <p></p>
    syntax:
      content: public class AuthorizationFailedException extends EventHubException
  - uid: com.microsoft.azure.eventhubs._communication_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._communication_exception.yml
    name: CommunicationException
    nameWithType: CommunicationException
    fullName: com.microsoft.azure.eventhubs.CommunicationException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-106">Diese Ausnahme wird ausgelöst, wenn ein Client-Side-Konnektivitätsproblem vorhanden ist.</span><span class="sxs-lookup"><span data-stu-id="760e8-106">This exception is thrown when there is a client side connectivity issue.</span></span> <span data-ttu-id="760e8-107">Konnektivität der Clienteinstellungen für den Dienst empfängt diese Ausnahme Benutzer wann überprüft werden soll:<itemizedlist><listitem></span><span class="sxs-lookup"><span data-stu-id="760e8-107">When receiving this exception user should check client connectivity settings to the service: <itemizedlist><listitem></span></span><p><span data-ttu-id="760e8-108">Überprüfen Sie die richtige Hostname und Portnummer, die im Endpunkt verwendet.</span><span class="sxs-lookup"><span data-stu-id="760e8-108">Check for correct hostname and port number used in endpoint.</span></span> </p></listitem><listitem><p><span data-ttu-id="760e8-109">Überprüfen Sie für alle möglichen Proxyeinstellungen, die Amqp-Ports zu blockieren, kann</span><span class="sxs-lookup"><span data-stu-id="760e8-109">Check for any possible proxy settings that can block amqp ports</span></span> </p></listitem><listitem><p><span data-ttu-id="760e8-110">Überprüfen Sie alle Firewalleinstellungen, die Amqp Ports blockieren können</span><span class="sxs-lookup"><span data-stu-id="760e8-110">Check for any firewall settings that can block amqp ports</span></span> </p></listitem><listitem><p><span data-ttu-id="760e8-111">Überprüfen Sie für jede allgemeine Probleme bei der Netzwerkverbindung sowie die Netzwerklatenz.</span><span class="sxs-lookup"><span data-stu-id="760e8-111">Check for any general network connectivity issues, as well as network latency.</span></span> </p></listitem></itemizedlist></p>
      <p></p>
    syntax:
      content: public class CommunicationException extends EventHubException
  - uid: com.microsoft.azure.eventhubs._illegal_entity_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._illegal_entity_exception.yml
    name: IllegalEntityException
    nameWithType: IllegalEntityException
    fullName: com.microsoft.azure.eventhubs.IllegalEntityException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-112">Diese Ausnahme tritt aus den folgenden Gründen:<itemizedlist><listitem></span><span class="sxs-lookup"><span data-stu-id="760e8-112">This exception is thrown for the following reasons: <itemizedlist><listitem></span></span><p><span data-ttu-id="760e8-113">Wenn die entitätsbenutzer versucht, eine Verbindung herzustellen, ist nicht vorhanden</span><span class="sxs-lookup"><span data-stu-id="760e8-113">When the entity user attempted to connect does not exist</span></span> </p></listitem><listitem><p><span data-ttu-id="760e8-114">Die Entität Benutzer möchte die Verbindung ist deaktiviert.</span><span class="sxs-lookup"><span data-stu-id="760e8-114">The entity user wants to connect is disabled</span></span> </p></listitem></itemizedlist></p>
      <p></p>
    syntax:
      content: public class IllegalEntityException
  - uid: com.microsoft.azure.eventhubs._operation_cancelled_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._operation_cancelled_exception.yml
    name: OperationCancelledException
    nameWithType: OperationCancelledException
    fullName: com.microsoft.azure.eventhubs.OperationCancelledException
    type: Class
    summary: '<p><span data-ttu-id="760e8-115">Diese Ausnahme wird ausgelöst, wenn die zugrunde liegenden Amqp-Ebene treffen, einen nicht ordnungsgemäßen Link abbrechen oder Trennen der Verbindung auf unerwartete Weise.</span><span class="sxs-lookup"><span data-stu-id="760e8-115">This exception is thrown when the underlying Amqp layer encounter an abnormal link abort or disconnect of connection in an unexpected fashion.</span></span> </p>'
    syntax:
      content: public class OperationCancelledException
  - uid: com.microsoft.azure.eventhubs._payload_size_exceeded_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._payload_size_exceeded_exception.yml
    name: PayloadSizeExceededException
    nameWithType: PayloadSizeExceededException
    fullName: com.microsoft.azure.eventhubs.PayloadSizeExceededException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-116">Diese Ausnahme wird ausgelöst, wenn der Benutzer versucht, Daten oder vermittelte Nachrichten, die die zulässige Nutzlastgröße gemäß der Definition durch den Dienst Zustellversuchen senden.</span><span class="sxs-lookup"><span data-stu-id="760e8-116">this exception is thrown when user attempts to send a event data or brokered message that has exceeded the allowed payload size as defined by the service.</span></span> <span data-ttu-id="760e8-117">Beachten Sie, die in einem Batch Szenario den Grenzwert senden kann mögliche Batch Aufwand einschließen.</span><span class="sxs-lookup"><span data-stu-id="760e8-117">Note that in a batch send scenario the limit can include possible batch overhead.</span></span></p>
      <p></p>
    syntax:
      content: public class PayloadSizeExceededException
  - uid: com.microsoft.azure.eventhubs._quota_exceeded_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._quota_exceeded_exception.yml
    name: QuotaExceededException
    nameWithType: QuotaExceededException
    fullName: com.microsoft.azure.eventhubs.QuotaExceededException
    type: Class
    syntax:
      content: public class QuotaExceededException
  - uid: com.microsoft.azure.eventhubs._receiver_disconnected_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._receiver_disconnected_exception.yml
    name: ReceiverDisconnectedException
    nameWithType: ReceiverDisconnectedException
    fullName: com.microsoft.azure.eventhubs.ReceiverDisconnectedException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-118">Diese Ausnahme wird ausgelöst, wenn eine EventHubReceiver aufgrund einer der folgenden Gründe getrennt wird:<itemizedlist><listitem></span><span class="sxs-lookup"><span data-stu-id="760e8-118">This exception is thrown when a EventHubReceiver is being disconnected because of one of the following reason: <itemizedlist><listitem></span></span><p><span data-ttu-id="760e8-119">Benutzer versucht, einen Empfänger nicht Epoche an einen Event Hub-Partition verbinden, wenn es ein Empfänger Epoche ist verbunden, der der Partition.</span><span class="sxs-lookup"><span data-stu-id="760e8-119">user attempts to connect a non-epoch receiver to a event hub partition, when there is an epoch receiver connected to the partition.</span></span> </p></listitem><listitem><p><span data-ttu-id="760e8-120">Verwenden Sie einen Empfänger Epoche für eine bestimmte Partition jedoch ein anderer Epoche Empfänger mit einem höheren Wert für den Beginn der Epoche eine Verbindung mit derselben Partition her.</span><span class="sxs-lookup"><span data-stu-id="760e8-120">you are using an epoch receiver for a given partition but another epoch receiver with a higher epoch value connects to the same partition.</span></span> </p><span data-ttu-id="760e8-121"></listitem></itemizedlist>Benutzer sollten entweder der gesamte Code nicht Epoche Empfänger verwenden, oder stellen Sie sicher, dass es ist nur eine Epoche Empfänger, die Verarbeitung einer bestimmten Partitions zu einem bestimmten Zeitpunkt zeitlich sicherstellen.</span><span class="sxs-lookup"><span data-stu-id="760e8-121"></listitem></itemizedlist> User should make sure either all code are using non-epoch receivers, or ensure that there is only one epoch receiver processing a given partition at any given point in time.</span></span></p>
      <p></p>
    syntax:
      content: public class ReceiverDisconnectedException
  - uid: com.microsoft.azure.eventhubs._server_busy_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._server_busy_exception.yml
    name: ServerBusyException
    nameWithType: ServerBusyException
    fullName: com.microsoft.azure.eventhubs.ServerBusyException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-122">Server ausgelastet Ausnahme wird ausgelöst, wenn die aktuelle Entität Aktivität übermäßigen Last auf dem Dienst eingefügt wurde.</span><span class="sxs-lookup"><span data-stu-id="760e8-122">Server busy exception is thrown when the current entity's activity has put excessive load onto the service.</span></span> <span data-ttu-id="760e8-123">Wenn diese Ausnahme festgestellt Benutzer sollte über mindestens 4 Sekunden erneut keine Wiederholung-Runtime-Vorgänge für die genannten Entität warten.</span><span class="sxs-lookup"><span data-stu-id="760e8-123">When encountered this exception user should wait at least 4 seconds before any retry/runtime operations for the said entity again.</span></span></p>
      <p></p>
    syntax:
      content: public class ServerBusyException
  - uid: com.microsoft.azure.eventhubs._timeout_exception
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._timeout_exception.yml
    name: TimeoutException
    nameWithType: TimeoutException
    fullName: com.microsoft.azure.eventhubs.TimeoutException
    type: Class
    summary: |-
      <p><span data-ttu-id="760e8-124">Diese Ausnahme wird ausgelöst, wenn der Vorgang der zuvor festgelegten Zeitlimit überschritten hat.</span><span class="sxs-lookup"><span data-stu-id="760e8-124">This exception is thrown when the operation has exceeded the predetermined time limit.</span></span> <span data-ttu-id="760e8-125">Benutzer sollte überprüfen, dass die Konnektivität zwischen Clientprozess und Dienst fehlerfrei ist.</span><span class="sxs-lookup"><span data-stu-id="760e8-125">User should check connectivity is healthy between client process and service.</span></span></p>
      <p></p>
    syntax:
      content: public class TimeoutException
  - uid: com.microsoft.azure.eventhubs._error_context
    parent: com.microsoft.azure.eventhubs
    href: com.microsoft.azure.eventhubs._error_context.yml
    name: ErrorContext
    nameWithType: ErrorContext
    fullName: com.microsoft.azure.eventhubs.ErrorContext
    type: Class
    syntax:
      content: public class ErrorContext